### ISSUE TITLE
Require settings `locale` property

### ISSUE THREAD

**Issue by drewdaemon** [elastic/elastic-charts#2208]
With https://github.com/elastic/elastic-charts/pull/2164 we got the ability to pass a locale to our charts. This was accomplished in Kibana with https://github.com/elastic/kibana/pull/168571.

However, `locale` is currently an optional property. This increases the risk that it will be overlooked when new charts invocations are added to Kibana.

We should consider requiring `locale` to make sure it is always provided.

**Comment by nickofthyme** [comment]
Good point, it looks like the `locale` prop is required on the `Settings` spec but the `Settings` spec is not required.

https://github.com/markov00/elastic-charts/blob/0427550c30ed282afd3480ed407d2c176c3d88fa/packages/charts/src/specs/settings.tsx#L621-L624

I don't think there is a current solution to fix this given how charts is enabled but I think the work in https://github.com/elastic/elastic-charts/pull/1424 to use a charts provider for exactly these type of cases where we want global chart defaults.
